nginx_rock/etc/nginx.conf

```diff
@@ -52,27 +52,280 @@ http {
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/search$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/user/.*/filter(/|$) {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/directory/room/.*$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/capabilities$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/keys/query$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/keys/changes$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/keys/claim$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/room_keys/ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/keys/upload/ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/login$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/register$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/register/available$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/v1/register/m.login.registration_token/validity$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(r0|v3|unstable)/password_policy$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/rooms/.*/redact {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/rooms/.*/send {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/rooms/.*/state/ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/rooms/.*/(join|invite|leave|ban|unban|kick)$ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/join/ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/knock/ {
+      include worker_location.conf;
+    }
+
+    location ~ ^/_matrix/client/(api/v1|r0|v3|unstable)/profile/ {
+      include worker_location.conf;
+    }
+
+    location /health {
+      access_log off;
+      add_header 'Content-Type' 'application/json';
+      return 204;
     }
 
     location  / {
-      proxy_read_timeout 300;
-      proxy_pass http://127.0.0.1:8008;
-      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-      proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
-      proxy_set_header Host $http_host;
-      proxy_set_header X-Real-IP $remote_addr;
-      client_max_body_size 50M;
-      proxy_http_version 1.1;
+      include main_location.conf;
```

Comment:
I'm wondering are these endpoints documented for the workers and if you can combine locations so it is not as long? Is there a reference for this config somewhere?